spec/support/controller_macros.rb

```diff
@@ -0,0 +1,5 @@
+module ControllerMacros
+  def mock_devise
+    @request.env['devise.mapping'] = Devise.mappings[:user]
```

Comment:
You may want to neste this in `before(:each)` as described in https://github.com/plataformatec/devise/wiki/How-To:-Test-controllers-with-Rails-3-and-4-%28and-RSpec%29. Otherwise looks good